src/20240422_Multi_MigrateRobotsToChainlinkAutomationV2/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol

```diff
@@ -0,0 +1,192 @@
+  }
+}
+
+/**
+ * @dev Deploy Optimism
+ * deploy-command: make deploy-ledger contract=src/20240422_Multi_MigrateRobotsToChainlinkAutomationV2/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol:DeployOptimism chain=optimism
+ * verify-command: npx catapulta-verify -b broadcast/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol/10/run-latest.json
+ */
+contract DeployOptimism is OptimismScript {
+  function run() external broadcast {
+    // deploy payloads
+    address payload0 = GovV3Helpers.deployDeterministic(
+      type(AaveV3Optimism_MigrateRobotsToChainlinkAutomationV2_20240422).creationCode
+    );
+
+    // compose action
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actions = new IPayloadsControllerCore.ExecutionAction[](1);
+    actions[0] = GovV3Helpers.buildAction(payload0);
+
+    // register action at payloadsController
+    GovV3Helpers.createPayload(actions);
+  }
+}
+
+/**
+ * @dev Deploy Arbitrum
+ * deploy-command: make deploy-ledger contract=src/20240422_Multi_MigrateRobotsToChainlinkAutomationV2/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol:DeployArbitrum chain=arbitrum
+ * verify-command: npx catapulta-verify -b broadcast/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol/42161/run-latest.json
+ */
+contract DeployArbitrum is ArbitrumScript {
+  function run() external broadcast {
+    // deploy payloads
+    address payload0 = GovV3Helpers.deployDeterministic(
+      type(AaveV3Arbitrum_MigrateRobotsToChainlinkAutomationV2_20240422).creationCode
+    );
+
+    // compose action
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actions = new IPayloadsControllerCore.ExecutionAction[](1);
+    actions[0] = GovV3Helpers.buildAction(payload0);
+
+    // register action at payloadsController
+    GovV3Helpers.createPayload(actions);
+  }
+}
+
+/**
+ * @dev Create Proposal
+ * command: make deploy-ledger contract=src/20240422_Multi_MigrateRobotsToChainlinkAutomationV2/MigrateRobotsToChainlinkAutomationV2_20240422.s.sol:CreateProposal chain=mainnet
+ */
+contract CreateProposal is EthereumScript {
+  function run() external {
+    // create payloads
+    PayloadsControllerUtils.Payload[] memory payloads = new PayloadsControllerUtils.Payload[](7);
+
+    // compose actions for validation
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsPolygonOne = new IPayloadsControllerCore.ExecutionAction[](1);
+    actionsPolygonOne[0] = GovV3Helpers.buildAction(
+      type(AaveV2Polygon_MigrateRobotsToChainlinkAutomationV2_20240422).creationCode
+    );
+    payloads[0] = GovV3Helpers.buildPolygonPayload(vm, actionsPolygonOne);
+
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsPolygonTwo = new IPayloadsControllerCore.ExecutionAction[](1);
+    actionsPolygonTwo[0] = GovV3Helpers.buildAction(
+      type(AaveV3Polygon_MigrateRobotsToChainlinkAutomationV2_20240422).creationCode
+    );
+    payloads[1] = GovV3Helpers.buildPolygonPayload(vm, actionsPolygonTwo);
```

Comment:
There can be two actions in one payload. It will ensure the order of execution

Reply:
We do not want the execution of both the payloads in this case in a single tx. So we execute payload 1 (which cancels the robot), then we need to wait 50 blocks and withdraw the link to collector and then execute payload 2.